### PR TITLE
[css-fonts] font-feature-settings and font-variation-settings should sort their tags alphabetically

### PR DESCRIPTION
**Pull request by graouts**
#### 39681d9b2bda037dcbaf89aa6e7c3ece7436d449
<pre>
[css-fonts] font-feature-settings and font-variation-settings should sort their tags alphabetically
<a href="https://bugs.webkit.org/show_bug.cgi?id=252238">https://bugs.webkit.org/show_bug.cgi?id=252238</a>

Reviewed by Myles C. Maxfield.

We have some WPT test failures for font-variation-settings animation due to our sorting duplicate
tags differently than Chrome and Firefox (alphabetically by ascending order). While the css-fonts
spec does not specify any order for tags in the computed style, we adopt the same style as Chrome
and Firefox for font-variation-settings.

Since this applies not just to animated values but this issue wasn&apos;t caught by parsing test, we add
some new WPT tests to css/css-fonts/parsing/font-variation-settings-computed.html.

Finally, since font-feature-settings is a similar type of property, and we implement both with the
templatized FontTaggedSettings class, we apply the same rule for this CSS property as well. We also
update our copy of the css/css-fonts/parsing/font-variation-settings-computed.html WPT since it now
passes with our new rule.

Since we implement the deduplication and sorting directly in FontTaggedSettings&lt;T&gt;::insert, there is
no need for the FontTaggedSettings&lt;T&gt;::deduplicated method used in ComputedStyleExtractor and
CSSPropertyAnimation which was recently introduced in 260212@main, so we largely revert that patch
as far as those two classes are concerned.

A spec issue was filed to match this behavior: <a href="https://github.com/w3c/csswg-drafts/issues/8450.">https://github.com/w3c/csswg-drafts/issues/8450.</a>

* LayoutTests/animations/font-variations/font-variation-settings-order-expected.txt:
* LayoutTests/animations/font-variations/font-variation-settings-order.html:
* LayoutTests/fast/text/variations/getComputedStyle-expected.txt:
* LayoutTests/fast/text/variations/getComputedStyle.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/inheritance-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-feature-settings-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-feature-settings-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-variation-settings-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-variation-settings-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-001-expected.txt:
* Source/WebCore/animation/CSSPropertyAnimation.cpp:
(WebCore::PropertyWrapperGetter::value const):
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::ComputedStyleExtractor::valueForPropertyInStyle):
* Source/WebCore/platform/graphics/FontTaggedSettings.h:
(WebCore::FontTaggedSettings&lt;T&gt;::insert):
(WebCore::FontTaggedSetting&lt;T&gt;::operator&lt; const): Deleted.
(WebCore::FontTaggedSettings&lt;T&gt;::deduplicated const): Deleted.

Canonical link: <a href="https://commits.webkit.org/260298@main">https://commits.webkit.org/260298@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/e93d04e49568f86b75c003b13104cc73bbd061cd

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/107736 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/16794 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/40638 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/116886 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/116272 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/111627 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/18213 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/8112 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/99918 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/113490 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/13755 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/96950 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/41438 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/95663 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/28590 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/83269 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/9765 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/29938 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/10448 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/6830 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/15919 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/49522 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/12010 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3865 "Built successfully and passed tests") | | | |

### CODE REVIEW

File: LayoutTests/TestExpectations [hunk 1]
```diff
@@ -3731,10 +3731,6 @@ webkit.org/b/182042 imported/w3c/web-platform-tests/css/css-fonts/font-default-0
 webkit.org/b/182042 imported/w3c/web-platform-tests/css/css-fonts/font-kerning-03.html [ ImageOnlyFailure ]
 webkit.org/b/86071 imported/w3c/web-platform-tests/css/css-fonts/font-kerning-05.html [ ImageOnlyFailure ]
 
-# font-variant-alternates: swash doesn't work
```
Comment (mdubet) [inline LayoutTests/TestExpectations:3734]: Thanks, this should have been rebased with E

Comment (mdubet) [inline LayoutTests/TestExpectations]: This change is unrelated to this patch it seems

Comment (graouts) [reply LayoutTests/TestExpectations]: I noticed that those tests were passing locally when they were expected not to, but I guess this has nothing to do with this patch. I'd remove this part when landing.

**Review by litherum — APPROVED** [review]

File: LayoutTests/TestExpectations (hunk 1 above)
Comment (graouts) [reply LayoutTests/TestExpectations]: I removed this.
